user/service.go

```diff
@@ -0,0 +1,78 @@
+package user
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type contextKeyType struct{}
+
+var (
+	// userIDContextKey is the key used for user.FromContext and
+	// user.NewContext.
+	userIDContextKey = contextKeyType(struct{}{})
+)
+
+// Service is a type of service that manages business process
+type Service struct {
+	repository Repository
+	config     Config
+}
+
+// ValidateUser check if user information is in header
+// if exist in header, validate with user info from DB, if not return error
+// if exist in DB, return user ID, if not exist in DB, create a new one
+func (s *Service) ValidateWithHeader(ctx context.Context, header http.Header) (string, error) {
```

Comment:
is it possible to move this validation on handlers layer 🤔 .
Not sure if we want to user service know http context.
This could impact for when we are introducing CLI interface.

Reply:
sure, let me check